src/librustdoc/config.rs

```diff
@@ -548,16 +551,25 @@ impl Options {
             dcx.fatal("the `--test` flag must be passed to enable `--no-run`");
         }
 
+        let mut output_to_stdout = false;
         let test_builder_wrappers =
             matches.opt_strs("test-builder-wrapper").iter().map(PathBuf::from).collect();
         let out_dir = matches.opt_str("out-dir").map(|s| PathBuf::from(&s));
-        let output = matches.opt_str("output").map(|s| PathBuf::from(&s));
+        let output = matches.opt_str("output");
         let output = match (out_dir, output) {
-            (Some(_), Some(_)) => {
-                dcx.fatal("cannot use both 'out-dir' and 'output' at once");
+            (Some(_), Some(output)) => {
+                if output == "-" {
+                    output_to_stdout = output == "-";
```

Comment:
NIT: This is always true by the condition above:

```suggestion
                    output_to_stdout = true;
```

Reply:
Oof, silly me. Thanks!